src/main/java/org/isf/examination/rest/ExaminationController.java

```diff
@@ -203,65 +204,80 @@ public ResponseEntity<List<PatientExaminationDTO>> getByPatientId(@PathVariable
 			return ResponseEntity.ok(listPatientExaminationDTO);
 		}
 	}
-	
-        public void validateExamination(PatientExaminationDTO newPatientExamination) throws OHServiceException {
-    	ExaminationParameters.initialize();
-		
-	if (newPatientExamination.getPex_height() < ExaminationParameters.HEIGHT_MIN || newPatientExamination.getPex_height() > ExaminationParameters.HEIGHT_MAX) {
-            throw new OHAPIException(new OHExceptionMessage("The size should be between "+ExaminationParameters.HEIGHT_MIN+" and " + ExaminationParameters.HEIGHT_MAX));
-        }
-        if (newPatientExamination.getPex_weight() < ExaminationParameters.WEIGHT_MIN || newPatientExamination.getPex_weight() > ExaminationParameters.WEIGHT_MAX) {
-            throw new OHAPIException(new OHExceptionMessage("The weight should be between"+ExaminationParameters.WEIGHT_MIN+" and " + ExaminationParameters.WEIGHT_MAX));
-        }
-        if (newPatientExamination.getPex_ap_min() != null && newPatientExamination.getPex_ap_max() != null) {
-            if (newPatientExamination.getPex_ap_min() > newPatientExamination.getPex_ap_max()) {
-            	throw new OHAPIException(new OHExceptionMessage("The minimum blood pressure must be lower than the maximum blood pressure"));
-            }
-        }
-        if (newPatientExamination.getPex_hr() != null) {
-            if (newPatientExamination.getPex_hr() < ExaminationParameters.HR_MIN || newPatientExamination.getPex_hr() > ExaminationParameters.HR_MAX) {
-            	throw new OHAPIException(new OHExceptionMessage("Heart rate should be between "+ExaminationParameters.HR_MIN +" and "+ExaminationParameters.HR_MAX));
-            }
-        }
-        if (newPatientExamination.getPex_temp() != null) {
-            if (newPatientExamination.getPex_temp() < ExaminationParameters.TEMP_MIN || newPatientExamination.getPex_temp() > ExaminationParameters.TEMP_MAX) {
-             	throw new OHAPIException(new OHExceptionMessage("The temperature should be between "+ExaminationParameters.TEMP_MIN +" and "+ExaminationParameters.TEMP_MAX));
-            }
-        }
-	if (newPatientExamination.getPex_sat() != null) {
-	    if (newPatientExamination.getPex_sat() < ExaminationParameters.SAT_MIN || newPatientExamination.getPex_sat() > ExaminationParameters.SAT_MAX) {
-	        throw new OHAPIException(new OHExceptionMessage("The saturation should be between "+ExaminationParameters.SAT_MIN+" and "+ExaminationParameters.SAT_MAX));
-	    } 
+
+	public void validateExamination(PatientExaminationDTO newPatientExamination) throws OHServiceException {
+		ExaminationParameters.initialize();
+
+		if (newPatientExamination.getPex_height() == null || newPatientExamination.getPex_weight() == null) {
```

Comment:
I would suggest getting height, weight, ap_min, hr, temp, etc. one time and put it into a local variable instead of fetching the values multiple times.

Reply:
I agree. Shall we put all of them at the top also?

Reply:
No; go get them just before you need them incase previous statements generate exceptions.

Reply:
so it's ok like that?